### DISCUSSION TITLE
How NuxtPage name prop working ?

### DISCUSSION THREAD

**Discussion by flihub** [nuxt/nuxt#19417] [Questions]
```
<NuxtPage name="about"  />
```
expect to render pages/about.vue component render ?

https://nuxt.com/docs/examples/routing/pages

![image](https://user-images.githubusercontent.com/16720735/222690419-781dbb4f-8aa4-4622-8e0d-1c013ddbb9b4.png)

**Comment by AloisSeckar** [comment]
In most scenarios I think you just leave `<NuxtPage>` as it is and don't use it with attributes at all. Because Nuxt auto-scans `/pages` directory and does all the job to fit in the correct component there.

`<NuxtPage>` is a wrapper around underlaying `<RouterView>` from [Vue Router](https://router.vuejs.org/). But altough it accepts `name` and `route` attributes, I failed to find an example of how to put them into good use. Even in Vue tutorials, they usually just place empty `<router-view>` tag inside template and do the actual routing with VueRouter.

Maybe enhancing https://nuxt.com/docs/api/components/nuxt-page with some example would be nice, if someone knows, how to do it.

**Comment by arkhaiel** [comment]
Hi, did you find anything more ?
I can't make it work either, I assumed it would work the same way as you did.

**Comment by urshofer** [comment]
Bump: Same Here. I got a simple use case where I'd like multiple named NuxtPages. Routes are written into different NuxtPages, and the state of each NuxtPage should be kept alive. No matter what I've tried it did not work. Never tried it with spa, because I need ssr.
